stable_baselines3/her/her_replay_buffer.py

```diff
@@ -375,12 +375,11 @@ def truncate_last_trajectory(self) -> None:
         If not called, we assume that we continue the same trajectory (same episode).
         """
         # If we are at the start of an episode, no need to truncate
-        if (self.ep_start[self.pos] != self.pos).any():
+        if (self._current_ep_start != self.pos).any():
             warnings.warn(
                 "The last trajectory in the replay buffer will be truncated.\n"
                 "If you are in the same episode as when the replay buffer was saved,\n"
                 "you should use `truncate_last_trajectory=False` to avoid that issue."
             )
-            self.ep_start[-1] = self.pos
-            # set done = True for current episodes
-            self.dones[self.pos - 1] = True
```

Comment:
If `self.dones[self.pos - 1] = False` before `truncate_last_trajectory`, the episode was not finished and `self.ep_length` is 0 for all entries belonging to this trajectory. Since `add()` is never called for this truncated trajectory, `self.ep_length[episode_indices]` is always 0 and cannot be sampled. So if I'm not mistaken, `self.dones[self.pos - 1] = True` has no effect.

Reply:
You're right. I guess the question is therefore: do we wan't the truncated episode to be sampled? To be consistent with the previous version, I would say yes. So we've to keep this line, but also update `self.ep_length[self.pos - 1]`

Reply:
What about infinite horizon tasks? If the episode is not finished before truncation, the done flag would become True too early.

Reply:
Good point. The infinite horizon case is handled with the timeouts buffer. We should also update it, something like

```python
if self.handle_timeout_termination:
    self.timeouts[self.pos-1] = True
```